sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationAsyncClient.java

```diff
@@ -1423,42 +1417,27 @@ public Mono<Response<TopicProperties>> updateTopicWithResponse(TopicProperties t
      */
     Mono<Response<QueueProperties>> createQueueWithResponse(String queueName, CreateQueueOptions createQueueOptions,
         Context context) {
-        if (queueName == null) {
-            return monoError(LOGGER, new NullPointerException("'queueName' cannot be null."));
-        } else if (queueName.isEmpty()) {
-            return monoError(LOGGER, new IllegalArgumentException("'queueName' cannot be empty."));
-        }
-
-        if (createQueueOptions == null) {
-            return monoError(LOGGER, new NullPointerException("'createQueueOptions' cannot be null."));
-        } else if (context == null) {
-            return monoError(LOGGER, new NullPointerException("'context' cannot be null."));
-        }
-        final Context contextWithHeaders = context.addData(AZ_TRACING_NAMESPACE_KEY, AZ_TRACING_NAMESPACE_VALUE)
-            .addData(AZURE_REQUEST_HTTP_HEADERS_KEY, new HttpHeaders());
-
-        final String forwardToEntity = createQueueOptions.getForwardTo();
-        if (!CoreUtils.isNullOrEmpty(forwardToEntity)) {
-            addSupplementaryAuthHeader(SERVICE_BUS_SUPPLEMENTARY_AUTHORIZATION_HEADER_NAME,
-                forwardToEntity, contextWithHeaders);
-            createQueueOptions.setForwardTo(getAbsoluteUrlFromEntity(forwardToEntity));
-        }
-
-        final String forwardDlqToEntity = createQueueOptions.getForwardDeadLetteredMessagesTo();
-        if (!CoreUtils.isNullOrEmpty(forwardDlqToEntity)) {
-            addSupplementaryAuthHeader(SERVICE_BUS_DLQ_SUPPLEMENTARY_AUTHORIZATION_HEADER_NAME,
-                forwardDlqToEntity, contextWithHeaders);
-            createQueueOptions.setForwardDeadLetteredMessagesTo(getAbsoluteUrlFromEntity(forwardDlqToEntity));
-        }
+        try {
+            validateQueueName(queueName);
+            if (createQueueOptions == null) {
+                throw LOGGER.logExceptionAsError(new NullPointerException("'createQueueOptions' cannot be null."));
```

Comment:
should return monoerror rather than throwing.

Reply:
Would it be fine, since this would be propogated to the monoError in the catch statement?